# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: architecture/sparc64/pkg_binutils.diff
# Copyright (C) 2006 - 2020 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

As GCC, we want the the linker to use elf32 by default, otherwise we would
have to patch packages like mozilla* or exact-image (and potentially other
exact-build based packages) that assume they can "pre"-link objects using
"ld -r". Of course the linker should accept the the default compiler output,
by default as well.

  - Rene Rebe <rene@exactcode.de>

--- binutils-2.34/ld/configure.tgt.vanilla	2020-02-05 19:12:07.555475478 +0100
+++ binutils-2.34/ld/configure.tgt	2020-02-05 19:13:23.855475515 +0100
@@ -843,10 +843,10 @@
 			targ_extra_libpath=$targ_extra_emuls
 			tdir_elf32_sparc=`echo ${targ_alias} | sed -e 's/64//'`
 			;;
-sparc64-*-linux-*)	targ_emul=elf64_sparc
-			targ_extra_emuls="elf32_sparc"
-			targ_extra_libpath=elf32_sparc
-			tdir_elf32_sparc=`echo ${targ_alias} | sed -e 's/64//'`
+sparc64-*-linux-*)	targ_emul=elf32_sparc
+			targ_extra_emuls="elf64_sparc"
+			targ_extra_libpath=elf64_sparc
+			tdir_elf32_sparc=`echo ${targ_alias} | sed -e 's/32//'`
 			;;
 sparc64-*-*bsd*)	targ_emul=elf64_sparc
 			targ_extra_emuls="elf32_sparc"
--- binutils-2.34/gas/configure.tgt.vanilla	2020-06-30 12:06:38.560577820 +0200
+++ binutils-2.34/gas/configure.tgt	2020-06-30 12:06:52.316577827 +0200
@@ -100,7 +100,7 @@
   score*)		cpu_type=score endian=big ;;
   sh*le)		cpu_type=sh endian=little ;;
   sh*)			cpu_type=sh endian=big ;;
-  sparc64* | sparcv9*)	cpu_type=sparc arch=v9-64 ;;
+  sparc64* | sparcv9*)	cpu_type=sparc arch=v8a ;;
   sparc86x*)		cpu_type=sparc arch=sparc86x  ;;
   sparclet*)		cpu_type=sparc arch=sparclet ;;
   sparclite*)		cpu_type=sparc arch=sparclite ;;
